pkg/client.go

```diff
@@ -35,6 +37,10 @@ type Client interface {
 	GenerateKeyPairFor(legalEntity types.LegalEntity) error
 	// SignFor signs a piece of data for a legal entity
 	SignFor(data []byte, legalEntity types.LegalEntity) ([]byte, error)
+	// SignCertificate issues a certificate by signing a PKCS10 certificate request. The private key of the specified CA should be available in the key store.
+	SignCertificate(entity types.LegalEntity, ca types.LegalEntity, pkcs10 []byte, profile CertificateProfile) ([]byte, error)
+	// GetPrivateKey returns the current private key for a given legal entity. It can be used for signing, but cannot be exported.
+	GetPrivateKey(entity types.LegalEntity) (crypto.Signer, error)
```

Comment:
Untill now private keys did not leave the crypto module, what's the use case?

Reply:
Because it is used by nuts-registry to sign a CSR. Go's CSR API requires an implementation of the Signer interface to sign the CSR.

Reply:
Renamed it to GetOpaquePrivateKey after discussion with @stevenvegt, since that's how Go calls it.